src/snowflake/cli/plugins/snowpark/package/anaconda.py

```diff
@@ -1,45 +1,79 @@
 from __future__ import annotations
 
 import logging
-from typing import List
+from typing import Dict, List, Set
 
 import requests
 from click import ClickException
-from packaging.version import parse
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement as PkgRequirement
+from packaging.version import InvalidVersion, parse
 from requests import HTTPError
-from snowflake.cli.plugins.snowpark.models import Requirement, SplitRequirements
+from snowflake.cli.plugins.snowpark.models import (
+    Requirement,
+    SplitRequirements,
+    WheelMetadata,
+)
 
 log = logging.getLogger(__name__)
 
 
+def _standarize_name(name: str) -> str:
+    return WheelMetadata.to_wheel_name_format(name.lower())
+
+
 class AnacondaChannel:
     snowflake_channel_url: str = (
         "https://repo.anaconda.com/pkgs/snowflake/channeldata.json"
     )
 
-    def __init__(self, packages):
-        self._packages = packages
+    def __init__(self, packages: Dict[str, Set[str]]):
+        """[packages] should be a dictionary mapping package name to set of its available versions"""
+        self._packages = {
+            _standarize_name(package_name): versions
+            for package_name, versions in packages.items()
+        }
 
     def is_package_available(
         self, package: Requirement, skip_version_check: bool = False
     ) -> bool:
-        package_name = package.name.lower()
+        package_name = _standarize_name(package.name)
         if package_name not in self._packages:
             return False
-        if package.specs and not skip_version_check:
-            latest_ver = parse(self._packages[package_name]["version"])
-            return all([parse(spec[1]) <= latest_ver for spec in package.specs])
-        return True
+        if skip_version_check or not package.specs:
+            return True
 
-    def package_version(self, package: Requirement):
-        return self._packages[package.name.lower()].get("version")
+        try:
+            package_specifiers = PkgRequirement(package.line).specifier
+            return any(
+                parse(version) in package_specifiers
+                for version in self._packages[package_name]
+            )
+        except (InvalidVersion, InvalidRequirement):
+            # fail-safe for non-pep508 formats
+            return False
+
+    def package_latest_version(self, package: Requirement) -> str | None:
+        package_name = _standarize_name(package.name)
+        if package_name not in self._packages:
+            return None
+        try:
+            versions = {parse(v) for v in self._packages[package_name]}
+        except InvalidVersion:
+            versions = self._packages[package_name]
+        return str(max(versions))
```

Comment:
Are we sure the max work properly here?

Reply:
For parsed versions - yes
For non-pep508 - might work not so good. The Anaconda example was `jpeg` package, which was available in versions `9c`, `9d` and `9e`.

As it is only used in in `snowpark package lookup`, I think good alternative might be simply list all possible versions for every package, WDYT?

Reply:
@sfc-gh-turbaszek changed in https://github.com/snowflakedb/snowflake-cli/pull/944/commits/45bd493fa94437d09da0ad48ef32e232c4758091